src/analysis/special_functions.rs

```diff
@@ -28,74 +32,151 @@ impl FromStr for Type {
             "free" | "destroy" => Ok(Free),
             "is_equal" => Ok(Equal),
             "ref" | "ref_" => Ok(Ref),
-            "to_string" => Ok(ToString),
+            "to_string" => Ok(Display),
             "unref" => Ok(Unref),
             "hash" => Ok(Hash),
             _ => Err(format!("Unknown type '{}'", s)),
         }
     }
 }
 
-pub type Infos = BTreeMap<Type, String>; // Type => glib_name
+impl Type {
+    fn extract(s: &str) -> Option<Self> {
+        s.parse().ok().or_else(|| match s {
+            "get_name" => Some(Self::Display),
+            _ => None,
+        })
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Info {
+    pub glib_name: String,
+    pub returns_static_ref: bool,
+    pub version: Option<Version>,
+}
+
+pub type Infos = BTreeMap<Type, Info>;
 
-fn update_func(func: &mut FuncInfo, type_: Type) -> bool {
+fn update_func(func: &mut FuncInfo, type_: Type, parent_type: &LibType) -> bool {
     if func.visibility != Visibility::Comment {
-        func.visibility = visibility(type_, func.parameters.c_parameters.len());
+        func.visibility = visibility(type_);
     }
-    // I assume `to_string` functions never return `NULL`
-    if type_ == Type::ToString {
-        if let Some(par) = func.ret.parameter.as_mut() {
-            *par.nullable = false;
+
+    if type_ == Type::Display {
+        if func.parameters.c_parameters.len() != 1 {
+            return false;
+        }
+        if !func.parameters.c_parameters[0].instance_parameter {
+            return false;
+        }
+        if !func
+            .ret
+            .parameter
+            .as_ref()
+            .map_or(false, |p| p.typ == TypeId::tid_utf8())
+        {
+            return false;
+        }
+
+        if func.name == "to_string" {
+            // Rename to to_str to make sure it doesn't clash with ToString::to_string
+            func.name = "to_str".to_owned();
+
+            // As to not change old code behaviour, assume non-nullability outside
+            // enums and flags only. Function inside enums and flags have been
+            // appropriately marked in Gir.
```

Comment:
This one should get linked to from https://github.com/gtk-rs/gtk-rs/issues/163

Also can you change it to only do that if the `trust_nullability` configuration is not set? If it is set then we better just trust it as configured by the user :)

Reply:
This comment has set up the link right now, is that enough?

I'll add a comment and see if I won't forget linking the exact line with a permalink, but that only makes sense once the commits are in master and are not going to be rebased anymore ;)

Reply:
Added `!trust_nullability` to the clause here, good idea but it changes nothing to neither gstreamer nor gtk :)